storage/fragmentationCache/fragmentationKeyData.hpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * \file
+ * \author Pavel Siska <siska@cesnet.cz>
+ * \brief Defines classes and structures for managing packet fragmentation data.
+ *
+ * This file contains the declarations of the `FragmentationKey`, `FragmentationData`,
+ * and `FragmentationKeyData` structures, along with a specialization of the
+ * `std::hash` template for the `FragmentationKey` structure. These structures
+ * are used for managing fragmented packet information and their associated data.
+ */
+/*
+ * Copyright (C) 2023 CESNET
+ *
+ * LICENSE TERMS
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name of the Company nor the names of its contributors
+ *    may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ */
+
+#pragma once
+
+#include "../xxhash.h"
+
+#include <cstring>
+#include <ctime>
+#include <functional>
+#include <ipfixprobe/ipaddr.hpp>
+#include <ipfixprobe/packet.hpp>
+
+namespace ipxp {
+
+/**
+ * @brief A struct representing a key for identifying fragmented packets.
+ *
+ * This struct is used to create keys for identifying fragmented packets based on
+ * their source IP, destination IP, fragmentation ID, and VLAN ID.
+ */
+struct FragmentationKey {
+    /**
+     * @brief Constructs a FragmentationKey object from a Packet structure.
+     *
+     * @param packet The Packet object from which to construct the key.
+     */
+    FragmentationKey(const Packet& packet)
+        : source_ip(packet.src_ip)
+        , destination_ip(packet.dst_ip)
+        , fragmentation_id(packet.frag_id)
+        , vlan_id(packet.vlan_id)
+    {
+    }
+
+    FragmentationKey() = default;
+
+    bool operator==(const FragmentationKey& other) const
+    {
+        if (std::memcmp(this, &other, sizeof(FragmentationKey)) == 0) {
+            return true;
+        }
+        return false;
+    }
+
+    ipaddr_t source_ip; ///< Source IP address of the packet.
```

Comment:
Shouldn't there be also a field that differentiates between ipv4 and ipv6? All ipv4 addresses could be interpreted as valid ipv6 addresses and this could create identical keys even when they shouldn't be same.